src/main/java/com/epam/eighty/service/impl/TopicServiceImpl.java

```diff
@@ -70,4 +71,33 @@ public Topic createTopic(final Topic topic, final Long id) {
         return topic;
     }
 
+    @Override
+    public Long getIdOfLastNotDeletedTopic(List<Long> topicIds) {
+        for(Long id: topicIds) {
+            Optional<Topic> topic = topicRepo.findOne(id);
+            if(topic.isPresent()) {
+                return id;
+            }
+        }
+        return null;
```

Comment:
I think it will be more beautiful and efficient to do this with stream.
It will be better return Optional instead of null.